.github/workflows/release.yaml

```diff
@@ -0,0 +1,29 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      issues: write
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2
+        with:
+          fetch-depth: 0
+      - name: Install dependencies
```

Comment:
Could we add a step here that sets up a specific Node release for the workflow? Something like this:

```
      - name: Setup NodeJS
        uses: actions/setup-node@v4 # note: commit hash pinning would be nice
        with:
          node-version: "22"
          check-latest: true
          cache: npm
```

This ensures the workflow is always run on the same Node version, despite any changes on the CI runner image.

Reply:
I had it roughly like that (well I used `"lts/*"` for the version) but thought it's kind of superfluous and we could get away with the minuscule maintenance burden and another workflow step it comes with.

But if we want the _same_ node version always, the above could use some tightening up. Or otherwise I suppose we could go with that `"lts/*"` instead of `"22"`.

No strong opinions though. But npm caching for sure is a nice thing to have.

Reply:
I've experienced issues caused by runner image changes every now and then. Hard to say how likely that is for this case, but it's quite low cost operation anyways. Definitely voting for an LTS version and I guess we could just pin it to the major version and use whatever minor version is available?

npm caching is a nice add but I guess these dependencies are quite small (in total). We get it for free though :)

Reply:
Deps... are not small :)
```shellsession
$ du -hc node_modules | tail -n 1
108M	total
```

In addition to that, the puppeteer dep grabs a local Chrome/Chromium during install, in a location I think we'll want to make sure to cache as well:
```shellsession
$ du -hc ~/.cache/puppeteer | tail -n 1
590.2M	total
```

Reply:
Alright, sounds like a good case for [actions/cache](https://github.com/actions/cache) if the installation path can't be changed to `.npm`?

Reply:
Nice!

Reply:
250+M cache compressed, so appears to have worked as expected: https://github.com/UpCloudLtd/upcloud-whmcs-module/actions/runs/13970833579/job/39111908917#step:11:4